src/providers/SignatureProvider.tsx

```diff
@@ -25,6 +25,7 @@ const createSignatureProviderState = () => {
     const [title, setTitle] = useState<InputElementOptions>(getDefaultInputElementOptions("title"));
     const [linkedin, setLinkedin] = useState<InputElementOptions>(getDefaultInputElementOptions("linkedin"));
     const [phone, setPhone] = useState<InputElementOptions>(getDefaultInputElementOptions("phone"));
+    const [valiEmail, setValiEmail] = useState<"valid" | "invalid">();
```

Comment:
🙈  okk, esté es un great example para _**know when you're going down a rabbit hole and ask for a second pair of eyes / brainstorm**_ 


Como la aplicación funciona es 

`Cambio en el input ---> triggers change in the provider state ---> change in provider state will be rendered in the table`


La manera de asegurarse de que la tabla no apareciera hasta que el email fuera valido era no actualizar el provider state


```javascript
// CODE BEFORE RABBIT HOLE REFACTOR: 
    const handleInputChange = (value: string) => {
        onInputChange({ variant: state.variant, textValue: value });
        if (state.variant === "email") {
            validateEmail(value);
        }
    };
```

```javascript
    const handleInputChange = (value: string) => {
        
        if (state.variant === "email") {
            cons isValid = validateEmail(value);
            // aca uno se asegura que el provider state solo va a cambiar si el email es válido
            if(isValid) {
              onInputChange({ variant: state.variant, textValue: value });
            }
        } else {
            onInputChange({ variant: state.variant, textValue: value });
        }
    };
```

Reply:
[Aquí está el git hash del commit](https://github.com/evargast/signature-generator/blob/b9ec140b519a4fdfa7e3aa3bb1b7c51129b06e5c/src/components/UsernameInput/UsernameInput.tsx) before the provider `setVali` refactor